sim/aviron/src/lib/memory.zig

```diff
@@ -0,0 +1,194 @@
+const std = @import("std");
+const io = @import("io.zig");
+
+/// Use the unified device interface
+pub const Backend = io.Device;
+const Device = io.Device;
+
+/// A mapping entry within a MemorySpace.
+pub const Segment = struct {
+    /// Base address within the enclosing memory space.
+    at: usize,
+    /// Size in bytes of the mapped range.
+    size: usize,
+    /// Backend handling the mapped range starting at index 0.
+    backend: Backend,
+};
+
+/// A logical memory space composed of non-overlapping segments.
+pub const MemorySpace = struct {
```

Comment:
I would call this `MemoryMapping` this is what it does:
It maps certain parts into a memory bus

Reply:
Done